src/main/scala/firrtl/passes/InferTypes.scala

```diff
@@ -66,16 +66,16 @@ object InferTypes extends Pass {
         types(s.name) = t
         s copy (tpe = t)
       case s: DefWire =>
-        val t = remove_unknowns(get_type(s))
+        val t = remove_unknowns(s.tpe)
         types(s.name) = t
         s copy (tpe = t)
       case s: DefNode =>
-        val sx = s map infer_types_e(types)
-        val t = remove_unknowns(get_type(sx))
+        val sx = s map infer_types_e(types) match { case sx: DefNode => sx }
```

Comment:
all this match does is result in a match error if the result is not a DefNode, is that what's intended?
If so, I think .asInstanceOf[DefNode] is actually more clear